beacon-chain/rpc/eth/beacon/state.go

```diff
@@ -148,6 +149,55 @@ func (bs *Server) GetFinalityCheckpoints(ctx context.Context, req *ethpb.StateRe
 	}, nil
 }
 
+// GetRandao fetches the RANDAO mix for the requested epoch from the state identified by state_id.
+// If an epoch is not specified then the RANDAO mix for the state's current epoch will be returned.
+// By adjusting the state_id parameter you can query for any historic value of the RANDAO mix.
+// Ordinarily states from the same epoch will mutate the RANDAO mix for that epoch as blocks are applied.
+func (bs *Server) GetRandao(ctx context.Context, req *eth2.RandaoRequest) (*eth2.RandaoResponse, error) {
+	ctx, span := trace.StartSpan(ctx, "beacon.GetRandao")
+	defer span.End()
+
+	var (
+		st  state.BeaconState
+		err error
+	)
+
+	st, err = bs.StateFetcher.State(ctx, req.StateId)
+	if err != nil {
+		return nil, helpers.PrepareStateFetchGRPCError(err)
+	}
+
+	var randao []byte
+	if req.Epoch == nil {
+		idx := uint64(st.RandaoMixesLength() - 1)
+		randao, err = st.RandaoMixAtIndex(idx)
+		if err != nil {
+			return nil, status.Errorf(codes.Internal, "Could not get randao mix at index %d", idx)
+		}
+	} else {
```

Comment:
I think the correct way to do this would be something like:
```
var epoch = slots.ToEpoch(st.Slot())
if req.Epoch != nil {
    epoch = *req.Epoch
}
idx := epoch % params.BeaconConfig().EpochsPerHistoricalVector
randao, err = st.RandaoMixAtIndex(idx)
if err != nil {
    return nil, status.Errorf(codes.Internal, "Could not get randao mix at index %d", idx)
}
```